src/auro-input.js

```diff
@@ -106,16 +114,125 @@ export default class AuroInput extends BaseInput {
     return null
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+
+    // Process auto-formating if defined
+    if (this.format) {
+      let config = null;
+
+      // Process only valid format strings
+      switch (this.format) {
+        case 'credit-card':
+          config = {
+            creditCard: true
+          }
+
+          break;
+        // add additional supported formats and their config JSON here
+        default:
+          // Do nothing
+      }
+
+      // initialize CleaveJS if we have a defined config for the requested format
+      if (config) {
+        // eslint-disable-next-line no-unused-vars
+        const cleave = new Cleave(this, config);
+      }
+    }
+  }
+
+  /**
+   * @private function to return html for credit card icon
+   * @returns {string} html string
+   */
+  getCreditCardIcon() {
+    // eslint-disable-next-line no-prototype-builtins
+    return this.icon && this.format && this.format === 'credit-card'
```

Comment:
```suggestion
    return this.icon && this.format === 'credit-card'
```

I am pretty sure that we don't need the additional validation of `this.format`.  This would be true for any of these in this doc.

Reply:
In VueJS I've always had to be explicit in this way so it became my habit to always do the pre-check for the attribute before the comparison. Looks like litElement doesn't complain in the same way so I'll remove the instances of this.